flaky test: `TestWithdrawal_Permissioned`

**Claude:** Reporting flaky test encountered in CI.

## Test
`TestWithdrawal_Permissioned` in `github.com/ethereum-optimism/optimism/op-acceptance-tests/tests/base/withdrawal/permissioned`

## CI Runs
- **memory-all-opn-op-geth:** https://app.circleci.com/pipelines/gh/ethereum-optimism/optimism/123586/workflows/3ab343a1-567f-44d3-a0e1-78b86d3b675b/jobs/4913300
  - Branch: merge-queue commit \`af44e3fa\` for #20343
  - Test ran 50.37s before failing

## Date
2026-04-24

## Failure
```
=== RUN   TestWithdrawal_Permissioned
    assertions.go:387:
        Error Trace:    op-devstack/dsl/eoa.go:191
                        op-devstack/dsl/eoa.go:181
                        op-devstack/dsl/bridge.go:182
                        op-acceptance-tests/tests/base/withdrawal/withdrawal_test_helper.go:63
        Error:          Received unexpected error:
                        upstream dep 0 (*plan.Lazy[*github.com/ethereum/go-ethereum/core/types.Receipt]) failed:
                        upstream dep 1 (*plan.Lazy[struct {}]) failed:
                        operation failed permanently after 5 attempts: nonce too low: next nonce 1, tx nonce 0
        Test:           TestWithdrawal_Permissioned
        Messages:       must transact
--- FAIL: TestWithdrawal_Permissioned (50.37s)
```

## Notes
- Failure looks like a nonce race in the EOA/transaction-management path (`op-devstack/dsl/eoa.go:191`). Two transactions sharing an account, the second submitted with stale nonce 0 while the first already advanced the chain to nonce 1.
- Encountered while merging an unrelated PR (#20343) which only touches `op-devstack/presets/singlechain_from_runtime.go` and `op-devstack/sysgo/{l2_cl_opnode,singlechain_build}.go` — none of those should affect withdrawal nonces.

A fix branch will follow.

--- a/op-service/txplan/txplan.go
+++ b/op-service/txplan/txplan.go
@@ -260,14 +260,20 @@ func WithTransactionSubmitter(cl TransactionSubmitter) Option {
 
 func WithRetrySubmission(cl TransactionSubmitter, maxAttempts int, strategy retry.Strategy) Option {
 	return func(tx *PlannedTx) {
-		tx.Submitted.DependOn(&tx.Signed)
+		// Intentionally no DependOn(&tx.Signed) here.
+		//
+		// The Lazy DAG evaluates a node's fn while holding read-locks on all of
+		// its upstream dependencies. If tx.Submitted declared tx.Signed as an
+		// upstream dep, the fn would run with tx.Signed's mutex read-locked.
+		// Calling tx.Signed.Eval or tx.Nonce.Invalidate (which cascades down to
+		// tx.Submitted) from inside that fn would then deadlock trying to
+		// acquire those same mutexes. Instead we hold no upstream locks and
+		// drive the full eval+submit cycle ourselves on each retry attempt.
 		tx.Submitted.Fn(func(ctx context.Context) (struct{}, error) {
 			return struct{}{}, retry.Do0(ctx, maxAttempts, strategy, func() error {
-				// Re-evaluate the signed tx on each attempt. On the first call
-				// this simply returns the already-evaluated value. On subsequent
-				// retries after a nonce-too-low error (see below) Nonce has been
-				// invalidated, so Eval will re-fetch the pending nonce and
-				// rebuild + re-sign the transaction.
+				// Evaluate (or re-evaluate after a nonce-too-low invalidation)
+				// the signed transaction. No upstream locks are held here, so
+				// Eval acquires the necessary mutexes safely.
 				signed, err := tx.Signed.Eval(ctx)
 				if err != nil {
 					return fmt.Errorf("re-evaluating signed tx: %w", err)

--- a/op-service/txplan/txplan.go
+++ b/op-service/txplan/txplan.go
@@ -14,6 +14,7 @@ import (
 	"github.com/ethereum/go-ethereum"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/params"
@@ -262,7 +263,25 @@ func WithRetrySubmission(cl TransactionSubmitter, maxAttempts int, strategy retr
 		tx.Submitted.DependOn(&tx.Signed)
 		tx.Submitted.Fn(func(ctx context.Context) (struct{}, error) {
 			return struct{}{}, retry.Do0(ctx, maxAttempts, strategy, func() error {
-				return cl.SendTransaction(ctx, tx.Signed.Value())
+				// Re-evaluate the signed tx on each attempt. On the first call
+				// this simply returns the already-evaluated value. On subsequent
+				// retries after a nonce-too-low error (see below) Nonce has been
+				// invalidated, so Eval will re-fetch the pending nonce and
+				// rebuild + re-sign the transaction.
+				signed, err := tx.Signed.Eval(ctx)
+				if err != nil {
+					return fmt.Errorf("re-evaluating signed tx: %w", err)
+				}
+				err = cl.SendTransaction(ctx, signed)
+				// If the nonce is stale (e.g. a deposit tx on L2 advanced the
+				// account nonce before we fetched it), invalidate the Nonce node
+				// so it is re-fetched on the next attempt. Because Nonce is
+				// upstream of Unsigned which is upstream of Signed, invalidating
+				// Nonce cascades through the whole signing pipeline.
+				if errors.Is(err, core.ErrNonceTooLow) {
+					tx.Nonce.Invalidate()
+				}
+				return err
 			})
 		})
 	}

